crypto/src/main/java/org/springframework/security/crypto/keygen/PkceCodeVerifierStringKeyGenerator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.crypto.keygen;
+
+import java.security.SecureRandom;
+import java.util.Base64;
+
+/**
+ * Creates a high-entropy cryptographic random string to use as the PKCE Code Verifier. By default,
+ * this uses 32 octets to produce a 43 character string--the minimum for a PKCE Code Verifier.
+ *
+ * <a target="_blank" href="https://tools.ietf.org/html/rfc7636#section-4.1">Section 4.1 Client Creates a Code Verifier</a>
+ * @since 5.2
+ * @author Stephen Doxsee
+ * @author Kevin Bolduc
+ */
+public class PkceCodeVerifierStringKeyGenerator implements StringKeyGenerator {
```

Comment:
Please remove this and use `Base64StringKeyGenerator` by supplying it a `Base64.Encoder` and `keyLength`. Also note that it uses `SecureRandom` internally.

Reply:
I'll try that out. Thanks